File: app/src/main/java/projects/csce/evence/di/AppComponent.java
package projects.csce.evence.di;

import javax.inject.Singleton;

import dagger.Component;
import projects.csce.evence.di.loggedin.LoggedInModule;
import projects.csce.evence.di.loggedin.LoggedInSubComponent;
import projects.csce.evence.view.main.MainActivity;

@Singleton
@Component(modules = {NetworkModule.class, LoginModule.class})
public interface AppComponent
{
    void inject(MainActivity mainActivity);

    LoggedInSubComponent newLoggedInSubComponent(LoggedInModule module);
}

File: app/src/main/java/projects/csce/evence/di/AppComponent.kt
package projects.csce.evence.di

import dagger.Component
import projects.csce.evence.di.loggedin.LoggedInModule
import projects.csce.evence.di.loggedin.LoggedInSubComponent
import projects.csce.evence.di.qr.QrModule
import projects.csce.evence.view.ui.GenerateQR
import projects.csce.evence.view.ui.MainActivity
import javax.inject.Singleton

@Singleton
@Component(modules = [NetworkModule::class, LoginModule::class, QrModule::class, ViewModelModule::class])
interface AppComponent
{
	fun inject(mainActivity: MainActivity)
	fun inject(generateQR: GenerateQR)
	fun newLoggedInSubComponent(module: LoggedInModule): LoggedInSubComponent
}